Description: Allow the setting of javac option using CMAKE_JAVA_COMPILE_FLAGS.
Author: Andrew Ross <andrewross@users.sourceforge.net>
Origin: upstream
Index: plplot-5.9.10/bindings/java/CMakeLists.txt
===================================================================
--- plplot-5.9.10.orig/bindings/java/CMakeLists.txt	2010-05-14 22:19:09.000000000 +0100
+++ plplot-5.9.10/bindings/java/CMakeLists.txt	2013-10-03 12:10:11.445188817 +0100
@@ -200,7 +200,7 @@
     add_custom_command(
       OUTPUT ${output_file}
       COMMAND ${CMAKE_Java_COMPILER} 
-      -classpath ${CMAKE_CURRENT_BINARY_DIR} ${srcfile} -d ${CMAKE_CURRENT_BINARY_DIR}
+      -source 1.5 -target 1.5 -classpath ${CMAKE_CURRENT_BINARY_DIR} ${srcfile} -d ${CMAKE_CURRENT_BINARY_DIR}
       DEPENDS ${srcfile} ${${output_file}_DEPENDS}
       )
   endforeach( srcfile ${JAVA_FILES_FULL} )
Index: plplot-5.9.10/examples/java/CMakeLists.txt
===================================================================
--- plplot-5.9.10.orig/examples/java/CMakeLists.txt	2012-10-03 22:16:31.000000000 +0100
+++ plplot-5.9.10/examples/java/CMakeLists.txt	2013-10-03 12:11:01.541187504 +0100
@@ -101,7 +101,7 @@
     add_custom_command(
       OUTPUT ${out_file}
       COMMAND ${CMAKE_Java_COMPILER}
-      -classpath ${CMAKE_BINARY_DIR}/bindings/java -d ${CMAKE_CURRENT_BINARY_DIR} -encoding UTF-8 ${in_file}
+      -source 1.5 -target 1.5 -classpath ${CMAKE_BINARY_DIR}/bindings/java -d ${CMAKE_CURRENT_BINARY_DIR} -encoding UTF-8 ${in_file}
       DEPENDS ${in_file} ${files_plplot_core}
       )
     add_custom_target(x${STRING_INDEX}j ALL DEPENDS ${out_file})
@@ -160,7 +160,7 @@
     add_custom_command(
       OUTPUT ${out_file}
       COMMAND ${CMAKE_Java_COMPILER}
-      -classpath ${INSTALLED_JAR_FILE} -d ${CMAKE_CURRENT_BINARY_DIR} -encoding UTF-8 ${in_file}
+      -source 1.5 -target 1.5 -classpath ${INSTALLED_JAR_FILE} -d ${CMAKE_CURRENT_BINARY_DIR} -encoding UTF-8 ${in_file}
       DEPENDS ${in_file} ${INSTALLED_JAR_FILE}
       )
     add_custom_target(x${STRING_INDEX}j ALL DEPENDS ${out_file})
